src/confluent_kafka/schema_registry/protobuf.py

```diff
@@ -238,6 +247,26 @@ def __init__(self, msg_type, schema_registry_client, conf=None):
         self._schema = Schema(_schema_to_str(descriptor.file),
                               schema_type='PROTOBUF')
 
+    @staticmethod
+    def _encode_varints(buf, ints):
+        """
+        Encodes each int as a varint onto buf
+
+        Args:
+            buf (BytesIO): buffer to write to.
+            ints ([int]): ints to be encoded.
+
+        """
+        for value in ints:
+            if(value < 0):
+                value = (-value) << 1 - 1
```

Comment:
This is not performing a shift since it is basically `(-value) << 0`.  Needs parens due to operator precedence.
Also; shouldn't this bug have been caught by a test?

Reply:
I suspect in practice this code is never used because we'll never be encoding negative values?

do you think it's better to test the negatives, or change the contract of the function so it errors on negative values?

Reply:
It'd be best if it was fixed, and tests added.